### PR TITLE
LPS-75593 After session timeout Social Ratings notifies user to sign in

### PR DESCRIPTION
**Pull request by fortunatomaldonado**
When the session times out and the user tries to rate a thread, the output will print "NaN". This is because the user does not have permission to leave a rating without being signed in.

The fix for this was checking when we receive the "NaN", which indicates the user is signed out, and graying out the rating icons. The user will no longer be able to click on the icons, but can see a tool tip stating to log in.

### CODE REVIEW

File: modules/apps/foundation/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/ratings.js [hunk 1]
```diff
@@ -496,59 +496,80 @@ AUI.add(
...
+							ratingThumbDown.attr('title', sessionTimeout);
+							ratingThumbDown.addClass(cssClasses.off);
+							ratingThumbDown.removeClass(cssClasses.on);
 
-						if (ratingThumbUp) {
-							ratingThumbUpCssClassOn = ratingThumbUp.hasClass(cssClassesOn);
+							instance.ratings.set('disabled', true);
 						}
+						else {
+
```
Comment (ericyanLr) [inline modules/apps/foundation/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/ratings.js]: Extra line inserted

File: modules/apps/foundation/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/ratings.js [hunk 2]
```diff
@@ -496,59 +496,80 @@ AUI.add(
...
+							var cssClassesOn = cssClasses.on;
+
+							var ratingThumbDown = elements.item(1);
```
Comment (ericyanLr) [inline modules/apps/foundation/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/ratings.js]: Redundant code, looks like it's been assigned above?

File: modules/apps/foundation/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/ratings.js [hunk 3]
```diff
@@ -496,59 +496,80 @@ AUI.add(
...
-						if (ratingThumbDown) {
-							if (ratingThumbDownCssClassOn) {
-								thumbDownMessage = Liferay.Language.get('you-have-rated-this-as-bad');
+							if (ratingThumbDown) {
+								ratingThumbDownCssClassOn = ratingThumbDown.hasClass(cssClassesOn);
 							}
-							else {
-								thumbDownMessage = Liferay.Language.get('rate-this-as-bad');
+
+							var ratingThumbUp = elements.item(0);
```
Comment (ericyanLr) [inline modules/apps/foundation/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/ratings.js]: Redundant code, looks like it's been assigned above?

**Comment by fortunatomaldonado** [comment]
@ericyanLr Must have missed them. Sorry about that, I'll fix it right now.

**Comment by ericyanLr** [comment]
Reviewed and sent to https://github.com/jonathanmccann/liferay-portal/pull/1452 for further review.

**Pull request closed without merging**